Couldn't connect to Discord via RPC:

I tried to use vscord but when trying to edit code it says:
`Couldn't connect to Discord via RPC: RPC_CONNECTION_TIMEOUT: Connection timed out`

I do have the necessary settings enabled in discord:
![image](https://user-images.githubusercontent.com/101394236/202863784-b6e03f30-9167-4b17-a83a-3b57c5af9442.png)

Heres the exact error log:
```js
[6:23:15 PM] [INFO] Discord Rich Presence for VS Code activated.
[6:23:15 PM] [INFO] Registered Discord Rich Presence commands
[6:23:25 PM] [ERROR] Encountered following error while trying to login: RPC_CONNECTION_TIMEOUT: Connection timed out
    at Ma.login (c:\Users\kuezy\.vscode-insiders\extensions\leonardssh.vscord-5.0.10\node_modules\@xhayper\discord-rpc\dist\Client.js:250:20)
    at new Qi (c:\Users\kuezy\.vscode-insiders\extensions\leonardssh.vscord-5.0.10\src\controller.ts:34:14)
    at Object.<anonymous> (c:\Users\kuezy\.vscode-insiders\extensions\leonardssh.vscord-5.0.10\src\extension.ts:10:20)
    at Module.u._compile (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\loader.js:4:1313)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1220:10)
    at Module.load (node:internal/modules/cjs/loader:1035:32)
    at Module._load (node:internal/modules/cjs/loader:876:12)
    at Function.c._load (node:electron/js2c/asar_bundle:5:13343)
    at Function.r._load (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:120:13363)
    at Function.w._load (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:115:61309)
    at Function.v._load (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:115:60713)
    at Module.require (node:internal/modules/cjs/loader:1059:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Function.r [as __$__nodeRequire] (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\loader.js:5:101)
    at d.rb (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:115:62572)
    at async Promise.all (index 0)
[6:23:25 PM] [INFO] [002] Destroyed Discord RPC client
[6:23:25 PM] [INFO] Successfully logged in to Discord with client ID 782685898163617802
```

When looking at the discord client ID it says `782685898163617802` which is not my client id...

Yep, got it. Here you go:

```
[6:19:34 PM] [INFO] [data.ts] ext(): Extension
[6:19:34 PM] [INFO] [data.ts] ext(): Changed to Extension
[6:19:34 PM] [INFO] [data.ts] [data.ts] ext(): Git extension is active
[6:19:34 PM] [INFO] [data.ts] api(): true
[6:19:34 PM] [INFO] [data.ts] api(): gitApi
[6:19:34 PM] [INFO] [data.ts] updateGit()
[6:19:34 PM] [INFO] [data.ts] gitRepoPath(): 
[6:19:34 PM] [INFO] [data.ts] updateGit(): repo 
[6:19:34 PM] [INFO] [data.ts] api(): true
[6:19:34 PM] [INFO] [data.ts] api(): gitApi
[6:19:34 PM] [INFO] [data.ts] updateGit()
[6:19:34 PM] [INFO] [data.ts] gitRepoPath(): 
[6:19:34 PM] [INFO] [data.ts] updateGit(): repo 
[6:19:34 PM] [INFO] Discord Rich Presence for VS Code activated.
[6:19:34 PM] [INFO] Registered Discord Rich Presence commands
[6:19:34 PM] [INFO] [data.ts] listeners(): Open Repo CUDA-Systembot
[6:19:34 PM] [INFO] [data.ts] updateGit()
[6:19:34 PM] [INFO] [data.ts] gitRepoPath(): d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:34 PM] [INFO] [data.ts] updateGit(): repo d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:34 PM] [INFO] [data.ts] listeners(): Open Repo CUDA-Systembot
[6:19:34 PM] [INFO] [data.ts] updateGit()
[6:19:34 PM] [INFO] [data.ts] gitRepoPath(): d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:34 PM] [INFO] [data.ts] updateGit(): repo d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:35 PM] [INFO] [data.ts] listeners(): Change State initialized
[6:19:35 PM] [INFO] [data.ts] updateGit()
[6:19:35 PM] [INFO] [data.ts] gitRepoPath(): d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:35 PM] [INFO] [data.ts] updateGit(): repo d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:35 PM] [INFO] [data.ts] listeners(): Change State initialized
[6:19:35 PM] [INFO] [data.ts] updateGit()
[6:19:35 PM] [INFO] [data.ts] gitRepoPath(): d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:35 PM] [INFO] [data.ts] updateGit(): repo d:\Users\Kuezy\04 Projects\[CUDA - DEVELOPMENT]\10 Develoment\00 System Builds\CUDA-Systembot
[6:19:44 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:19:44 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:19:44 PM] [ERROR] Encountered following error while trying to login: RPC_CONNECTION_TIMEOUT: Connection timed out
    at za.login (c:\Users\kuezy\.vscode-insiders\extensions\leonardssh.vscord-5.0.11\node_modules\@xhayper\discord-rpc\dist\Client.js:250:20)
    at new en (c:\Users\kuezy\.vscode-insiders\extensions\leonardssh.vscord-5.0.11\src\controller.ts:34:14)
    at Object.<anonymous> (c:\Users\kuezy\.vscode-insiders\extensions\leonardssh.vscord-5.0.11\src\extension.ts:10:20)
    at Module.u._compile (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\loader.js:4:1313)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1220:10)
    at Module.load (node:internal/modules/cjs/loader:1035:32)
    at Module._load (node:internal/modules/cjs/loader:876:12)
    at Function.c._load (node:electron/js2c/asar_bundle:5:13343)
    at Function.r._load (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:120:13370)
    at Function.w._load (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:115:61374)
    at Function.v._load (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:115:60776)
    at Module.require (node:internal/modules/cjs/loader:1059:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Function.r [as __$__nodeRequire] (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\loader.js:5:101)
    at d.rb (e:\Users\Kuezy\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:115:62638)
    at async Promise.all (index 0)
[6:19:44 PM] [INFO] [002] Destroyed Discord RPC client
[6:19:44 PM] [INFO] Successfully logged in to Discord with client ID 782685898163617802
[6:19:57 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:19:57 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:19:59 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:19:59 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:20:00 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:20:00 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:20:13 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:20:13 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:21:49 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:21:49 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor: got unallowed scheme, got 'output'
[6:21:49 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:21:51 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor
[6:21:51 PM] [INFO] [data.ts] root(): window.onDidChangeActiveTextEditor: got unallowed scheme, got 'output'
```

--- a/src/controller.ts
+++ b/src/controller.ts
@@ -111,22 +111,20 @@ export class RPCController {
         if (config.get(CONFIG_KEYS.Status.Idle.Check)) this.listeners.push(changeWindowState);
 
         this.listeners.push(fileSwitch, fileEdit, fileSelectionChanged, debugStart, debugEnd, gitListener);
-
     }
     private checkCanSend(): boolean {
         const config = getConfig();
         let userId = this.client.user?.id;
-        if(!userId) return false;
+        if (!userId) return false;
         let whitelistEnabled = config.get(CONFIG_KEYS.App.WhitelistEnabled);
-        if(whitelistEnabled) {
+        if (whitelistEnabled) {
             let whitelist = config.get(CONFIG_KEYS.App.Whitelist);
-            if(config.get(CONFIG_KEYS.App.whitelistIsBlacklist))
-                if(whitelist!.includes(userId)) return this.canSendActivity = false;
-                else return this.canSendActivity = true;
-            else
-                if(!whitelist!.includes(userId)) return this.canSendActivity = false;
+            if (config.get(CONFIG_KEYS.App.whitelistIsBlacklist))
+                if (whitelist!.includes(userId)) return (this.canSendActivity = false);
+                else return (this.canSendActivity = true);
+            else if (!whitelist!.includes(userId)) return (this.canSendActivity = false);
         }
-        return this.canSendActivity = true;
+        return (this.canSendActivity = true);
     }
 
     private async checkIdle(windowState: WindowState) {
@@ -176,6 +174,7 @@ export class RPCController {
         if (!this.enabled) return;
         this.checkCanSend();
         this.state = await activity(this.state, isViewing, isIdling);
+        this.state.instance = true;
         if (!this.state || Object.keys(this.state).length === 0 || !this.canSendActivity)
             return void this.client.user?.clearActivity(process.pid);
         return this.client.user?.setActivity(this.state, process.pid);

--- a/src/data.ts
+++ b/src/data.ts
@@ -15,7 +15,7 @@ import {
     workspace
 } from "vscode";
 
-const ALLOWED_SCHEME = ["file", "vscode-remote"];
+const ALLOWED_SCHEME = ["file", "vscode-remote", "untitled"];
 
 interface DisposableLike {
     dispose: () => unknown;

